### PR TITLE
use junit5 to test all the modules

### PR DESCRIPTION
**Pull request by lanwen**
also:
- bump testcontainers to 1.15.3 (as it can't find old ryuk)

### CODE REVIEW

File: examples/java/src/main/java/com/example/Consumer.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public static void main(String[] args) {
     }
 
     private static String getLiiklusTarget() {
-        GenericContainer<?> liiklus = new GenericContainer<>("bsideup/liiklus:0.9.0")
+        GenericContainer<?> liiklus = new GenericContainer<>("bsideup/liiklus:latest")
```
Comment (bsideup) [inline examples/java/src/main/java/com/example/Consumer.java:97]: let's not use `:latest` here - may break the build unexpectedly

File: examples/plugin/build.gradle [hunk 2]
```diff
@@ -38,11 +34,11 @@ dependencyManagement {
 
     imports {
         mavenBom org.springframework.boot.gradle.plugin.SpringBootPlugin.BOM_COORDINATES
-        mavenBom 'org.testcontainers:testcontainers-bom:1.12.3'
+        mavenBom 'org.testcontainers:testcontainers-bom:1.15.3'
     }
 
     dependencies {
-        dependencySet(group: "com.github.bsideup.liiklus", version: "0.9.0") {
+        dependencySet(group: "com.github.bsideup.liiklus", version: "0.10.0-rc1") {
```
Comment (bsideup) [inline examples/plugin/build.gradle:41]: outside of the scope of this PR

File: examples/plugin/src/main/java/com/github/bsideup/liiklus/plugins/example/ExampleRecordPostProcessor.java [hunk 3]
```diff
@@ -15,11 +16,12 @@
...
-                                record.getEnvelope().withValue(ByteBuffer.wrap("**masked**".getBytes())),
-                                record.getTimestamp(),
-                                record.getPartition(),
-                                record.getOffset()
+                        LiiklusCloudEvent value = (LiiklusCloudEvent) record.getEnvelope().getRawValue();
+                        return record.withEnvelope(record.getEnvelope()
+                                .withValue(
+                                        value.withData(ByteBuffer.wrap("**masked**".getBytes())),
+                                        LiiklusCloudEvent::asJson
+                                )
```
Comment (bsideup) [inline examples/plugin/src/main/java/com/github/bsideup/liiklus/plugins/example/ExampleRecordPostProcessor.java:24]: unrelated to this PR

File: examples/plugin/src/main/java/com/github/bsideup/liiklus/plugins/example/ExampleRecordPreProcessor.java [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+        LiiklusCloudEvent event;
+        if (envelope.getRawValue() instanceof LiiklusCloudEvent) {
+            event = (LiiklusCloudEvent) envelope.getRawValue();
+        } else {
+            CompletableFuture<Envelope> completableFuture = new CompletableFuture<>();
+            completableFuture.completeExceptionally(new IllegalArgumentException("raw value have to be LiiklusCloudEvent, got: " + envelope.getRawValue().getClass().getName()));
+            return completableFuture;
+        }
+
+        String value = event.getData().map(StandardCharsets.UTF_8::decode).orElse(CharBuffer.wrap("empty-string")).toString();
```
Comment (bsideup) [inline examples/plugin/src/main/java/com/github/bsideup/liiklus/plugins/example/ExampleRecordPreProcessor.java:28]: unrelated to this PR

File: examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/SmokeTest.java [hunk 5]
```diff
@@ -1,39 +1,39 @@
...
         String key = UUID.randomUUID().toString();
 
         publishRecord(key, "Hello!");
 
-        Record record = receiveRecords(key).blockFirst(Duration.ofSeconds(10));
+        ReceiveReply.LiiklusEventRecord record = receiveRecords(key).blockFirst(Duration.ofSeconds(10));
 
         assertThat(record).isNotNull().satisfies(it -> {
-            assertThat(it.getValue().toStringUtf8()).isEqualTo("!olleH");
+            assertThat(it.getEvent().getData().toStringUtf8()).isEqualTo("!olleH");
```
Comment (bsideup) [inline examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/SmokeTest.java:23]: unrelated change

File: examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/support/AbstractIntegrationTest.java [hunk 6]
```diff
@@ -45,21 +51,36 @@ protected PublishReply publishRecord(String key, String value) {
...
+                        LiiklusEvent.newBuilder()
+                                .setId(UUID.randomUUID().toString())
+                                .setType("com.example.event")
+                                .setSource("/tests")
+                                .setDataContentType("application/json")
+                                .putExtensions("comexampleextension1", "foo")
+                                .setData(ByteString.copyFromUtf8(value))
+                                .setTime(ZonedDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME))
+                                .buildPartial()
+                )
```
Comment (bsideup) [inline examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/support/AbstractIntegrationTest.java:64]: unrelated to this PR

Comment (lanwen) [reply examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/support/AbstractIntegrationTest.java:64]: yeah, yeah, I've got it :) will split as well.

File: examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/support/AbstractIntegrationTest.java [hunk 7]
```diff
@@ -13,14 +17,16 @@
...
+import java.time.format.DateTimeFormatter;
 import java.util.UUID;
 
 public abstract class AbstractIntegrationTest {
 
     protected static final LiiklusClient client;
 
     static {
-        LiiklusContainer liiklus = new LiiklusContainer("0.7.0")
+        LiiklusContainer liiklus = new LiiklusContainer(LiiklusContainer.class.getPackage().getImplementationVersion())
```
Comment (bsideup) [inline examples/plugin/src/test/java/com/github/bsideup/liiklus/plugins/example/support/AbstractIntegrationTest.java:29]: unrelated, also I would not use "always the latest" here

File: testing/src/main/java/com/github/bsideup/liiklus/container/LiiklusContainer.java [hunk 8]
```diff
@@ -17,7 +17,8 @@ public LiiklusContainer(String version) {
     }
 
     public LiiklusContainer withKafka(KafkaContainer kafkaContainer) {
-        return withKafka(kafkaContainer.getNetwork(), kafkaContainer.getNetworkAliases().get(0) + ":9092");
+        return dependsOn(kafkaContainer)
+                .withKafka(kafkaContainer.getNetwork(), "PLAINTEXT://" + kafkaContainer.getNetworkAliases().get(0) + ":9093");
```
Comment (bsideup) [inline testing/src/main/java/com/github/bsideup/liiklus/container/LiiklusContainer.java:21]: unrelated change

File: testing/src/test/java/com/github/bsideup/liiklus/container/LiiklusContainerTest.java [hunk 9]
```diff
@@ -1,29 +1,30 @@
...
+import org.junit.jupiter.api.Test;
 import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
 
 public class LiiklusContainerTest {
 
-    static final String LATEST_VERSION = "0.7.0";
-    static KafkaContainer kafka = new KafkaContainer();
+    static final String LATEST_VERSION = "latest";
```
Comment (bsideup) [inline testing/src/test/java/com/github/bsideup/liiklus/container/LiiklusContainerTest.java:10]: no `:latest`

**Review by bsideup — APPROVED** [review]
💯
